crash with 'panic: runtime error: invalid memory address or nil pointer dereference'

**Describe the bug**
crash with 'panic: runtime error: invalid memory address or nil pointer dereference', when click `share screen` more than once.

**Server**
- Version: [1.2.1]
- Environment: [docker on centos 7]

**Client**
- SDK: [react]
- Version: [1.3.2]

**To Reproduce**
Steps to reproduce the behavior:
1. one client is connected to room
2. click `share screen` , then click `stop share`
3. another client connect to this room.
4. crash...
it's repeat 100%.

**Additional context**
The log details are as follows
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xd5aa14]

goroutine 4231 [running]:
github.com/livekit/livekit-server/pkg/rtc.(*MediaTrackSubscriptions).AddSubscriber.func1()
	/workspace/pkg/rtc/mediatracksubscriptions.go:142 +0x1d4
github.com/livekit/livekit-server/pkg/sfu.(*DownTrack).Bind(0xc000822280, {{0xc0029f8000, 0x20}, {{0xc0022725a0, 0x2, 0x2}, {0xc0016c1b80, 0x4, 0x4}}, 0x24c351e2, ...})
	/workspace/pkg/sfu/downtrack.go:298 +0x68a
github.com/pion/webrtc/v3.(*RTPSender).Send(0xc003c10100, {{{0xc002272570, 0x2, 0x2}, {0xc0016c1b80, 0x4, 0x4}}, {0xc002d5e080, 0x1, 0x1}})
	/go/pkg/mod/github.com/pion/webrtc/v3@v3.1.43/rtpsender.go:305 +0x515
github.com/pion/webrtc/v3.(*PeerConnection).startRTPSenders(0x1024820?, {0xc002272540, 0x5, 0xc00034b6c0?})
	/go/pkg/mod/github.com/pion/webrtc/v3@v3.1.43/peerconnection.go:1386 +0x1bd
github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription(0xc003253440, {0x3, {0xc0018f0a00, 0x19c8}, 0xc001d885a0})
	/go/pkg/mod/github.com/pion/webrtc/v3@v3.1.43/peerconnection.go:1146 +0x6ef
github.com/livekit/livekit-server/pkg/rtc.(*PCTransport).setRemoteDescription(0xc001e1d500, {0x100000000000000?, {0xc0018eba80?, 0xc001e1d618?}, 0x0?})
	/workspace/pkg/rtc/transport.go:1657 +0x3ad
github.com/livekit/livekit-server/pkg/rtc.(*PCTransport).handleRemoteAnswerReceived(0xc001e1d500, 0xc0045bb5c8?)
	/workspace/pkg/rtc/transport.go:1758 +0x34
github.com/livekit/livekit-server/pkg/rtc.(*PCTransport).handleRemoteDescriptionReceived(0xc000871f80?, 0x1?)
	/workspace/pkg/rtc/transport.go:1626 +0x49
github.com/livekit/livekit-server/pkg/rtc.(*PCTransport).handleEvent(0xc000b7e010?, 0xd57f60?)
	/workspace/pkg/rtc/transport.go:1308 +0xbe
github.com/livekit/livekit-server/pkg/rtc.(*PCTransport).processEvents(0xc001e1d500)
	/workspace/pkg/rtc/transport.go:1281 +0xce
created by github.com/livekit/livekit-server/pkg/rtc.NewPCTransport
	/workspace/pkg/rtc/transport.go:343 +0x256

```

--- a/pkg/rtc/mediatrackreceiver.go
+++ b/pkg/rtc/mediatrackreceiver.go
@@ -304,9 +304,11 @@ func (t *MediaTrackReceiver) TryClose() bool {
 		return true
 	}
 
-	if len(t.receiversShadow) > 0 {
-		t.lock.RUnlock()
-		return false
+	for _, receiver := range t.receiversShadow {
+		if dr, _ := receiver.TrackReceiver.(*DummyReceiver); dr != nil && dr.Receiver() != nil {
+			t.lock.RUnlock()
+			return false
+		}
 	}
 	t.lock.RUnlock()
 	t.Close()
@@ -482,7 +484,8 @@ func (t *MediaTrackReceiver) addSubscriber(sub types.LocalParticipant) (err erro
 		streamId = PackStreamID(t.PublisherID(), t.ID())
 	}
 
-	err = t.MediaTrackSubscriptions.AddSubscriber(sub, NewWrappedReceiver(receivers, t.ID(), streamId, potentialCodecs))
+	tLogger := LoggerWithTrack(sub.GetLogger(), t.ID(), t.params.IsRelayed)
+	err = t.MediaTrackSubscriptions.AddSubscriber(sub, NewWrappedReceiver(receivers, t.ID(), streamId, potentialCodecs, tLogger))
 	if err != nil {
 		return
 	}

--- a/pkg/rtc/wrappedreceiver.go
+++ b/pkg/rtc/wrappedreceiver.go
@@ -9,6 +9,7 @@ import (
 	"go.uber.org/atomic"
 
 	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
 
 	"github.com/livekit/livekit-server/pkg/sfu"
 )
@@ -22,9 +23,11 @@ type WrappedReceiver struct {
 	streamId        string
 	codecs          []webrtc.RTPCodecParameters
 	determinedCodec webrtc.RTPCodecCapability
+
+	logger logger.Logger
 }
 
-func NewWrappedReceiver(receivers []*simulcastReceiver, trackID livekit.TrackID, streamId string, upstreamCodecs []webrtc.RTPCodecParameters) *WrappedReceiver {
+func NewWrappedReceiver(receivers []*simulcastReceiver, trackID livekit.TrackID, streamId string, upstreamCodecs []webrtc.RTPCodecParameters, logger logger.Logger) *WrappedReceiver {
 	sfuReceivers := make([]sfu.TrackReceiver, 0, len(receivers))
 	for _, r := range receivers {
 		sfuReceivers = append(sfuReceivers, r.TrackReceiver)
@@ -44,6 +47,7 @@ func NewWrappedReceiver(receivers []*simulcastReceiver, trackID livekit.TrackID,
 		trackID:   trackID,
 		streamId:  streamId,
 		codecs:    codecs,
+		logger:    logger,
 	}
 }
 
@@ -65,6 +69,12 @@ func (r *WrappedReceiver) DetermineReceiver(codec webrtc.RTPCodecCapability) {
 			// audio opus/red can match opus only
 			r.TrackReceiver = receiver.GetPrimaryReceiverForRed()
 			break
+		}
+	}
+	if r.TrackReceiver == nil {
+		r.logger.Errorw("can't determine receiver for codec", nil, "codec", codec.MimeType)
+		if len(r.receivers) > 0 {
+			r.TrackReceiver = r.receivers[0]
 		}
 	}
 }

